src/darker/git.py

```diff
@@ -207,15 +208,49 @@ def _git_check_output_lines(
     cmd: List[str], cwd: Path, exit_on_error: bool = True
 ) -> List[str]:
     """Log command line, run Git, split stdout to lines, exit with 123 on error"""
+    return _git_check_output(
+        cmd,
+        cwd,
+        exit_on_error=exit_on_error,
+        encoding="utf-8",
+    ).splitlines()
+
+
+@overload
+def _git_check_output(
+    cmd: List[str], cwd: Path, exit_on_error: bool = ..., encoding: None = ...
+) -> bytes:
+    ...
+
+
+@overload
+def _git_check_output(
+    cmd: List[str], cwd: Path, exit_on_error: bool, encoding: str
+) -> str:
+    ...
+
+
+@overload
+def _git_check_output(cmd: List[str], cwd: Path, *, encoding: str) -> str:
+    ...
```

Comment:
I guess there's no way around having to define these both, instead of just one `-> str` variant wtih a default value for `exit_on_error`?